MINIBALL: add Device virtualization docs

yaoyinnan opened the pull request:
Add description document for Device virtualization part. Modify the description of Device virtualization in MiniBall's README.md.

Fixes: https://github.com/openanolis/dragonball-sandbox/issues/204

Signed-off-by: yaoyinnan <yaoyinnan@foxmail.com>

studychao commented on crates/dbs-miniball/docs/DeviceVirtualization.md:
```diff
@@ -0,0 +1,307 @@
+- 将串行端口添加到总线（PIO 或 MMIO）
+- 事件处理（可选）
+
+第 6 ～ 9 行，创建 serial console。第 13 ～ 15 行，设置 irq 为 COM1_IRQ = 4，console manager 的 irq 为 3 和 4。第 17 ～ 20 行，要通过serial类型的console进入虚拟机的console，需要在虚拟机的cmdline中添加console=ttyS0。第 24 ～ 27 行，创建 PioAddressRange 定义串行端口的地址和大小。第 29 ～ 33 行，注册 serial console。第 36 行，通过 evnet manager 监听事件处理。
```
`evnet manager` typo

yaoyinnan replied:
sorry，fixed

studychao commented on crates/dbs-miniball/docs/DeviceVirtualization.md:
````diff
@@ -0,0 +1,307 @@
+```
+
+### 键盘 i8042
+I8042 数据端口（IO 端口 0x60）用于读取从 I8042 设备或 I8042 控制器本身接收的数据，并将数据写入 I8042 设备或 I8042 控制器本身。第 11 行，设置 irq 为 1，具有非常高的优先级。第 13 ～ 18 行，0x60 和 0x64 是 i8042 设备使用的 io 端口。将 pio 地址范围从 0x60 - 0x64 注册到 I8042_DATA_PORT 以供 i8042 使用。
````
Please mention why we need i8042

yaoyinnan replied:
added

studychao commented on crates/dbs-miniball/docs/DeviceVirtualization.md:
````diff
@@ -0,0 +1,307 @@
+}
+```
+
+## virtio 设备
````
vmm-reference has virtio-net, would you cover this part?

yaoyinnan replied:
Sorry, I haven't implemented virtio net in Miniball yet.

studychao commented on crates/dbs-miniball/docs/DeviceVirtualization.md:
````diff
@@ -0,0 +1,307 @@
+```
+
+### 键盘 i8042
+I8042 数据端口（IO 端口 0x60）用于读取从 I8042 设备或 I8042 控制器本身接收的数据，并将数据写入 I8042 设备或 I8042 控制器本身。第 11 行，设置 irq 为 1，具有非常高的优先级。第 13 ～ 18 行，0x60 和 0x64 是 i8042 设备使用的 io 端口。将 pio 地址范围从 0x60 - 0x64 注册到 I8042_DATA_PORT 以供 i8042 使用。在 Miniball 中，i8042 PS/2 控制器仅模拟 CPU 复位命令，该命令用于通知 VMM 来宾关闭。
````
`宾关闭`? typo?

yaoyinnan replied:
fixed

yaoyinnan committed:
MINIBALL: add Device virtualization docs

Add description document for Device virtualization part. Modify the description of Device virtualization in MiniBall's README.md.

Fixes: https://github.com/openanolis/dragonball-sandbox/issues/204

Signed-off-by: yaoyinnan <yaoyinnan@foxmail.com>

studychao committed:
Merge branch 'main' into 204/miniball/device-virtualization-docs
